### ISSUE TITLE
Parser too forgiving with generator expressions as arguments to functions

### ISSUE THREAD

**Issue by danr** [microsoft/pyright#2932]
**To Reproduce**
```
$ cat z.py 
foo = min(x for x in [0,1], default=-1)
$ python z.py 
  File "/tmp/z/z.py", line 1
    foo = min(x for x in [0,1], default=-1)
              ^^^^^^^
SyntaxError: Generator expression must be parenthesized
$ pyright z.py 
No configuration file found.
No pyproject.toml file found.
stubPath /tmp/z/typings is not a valid directory.
Assuming Python platform Linux
Searching for source files
File or directory "/tmp/z/--" does not exist.
Found 1 source file
0 errors, 0 warnings, 0 informations 
Completed in 0.571sec
```

**Expected behavior**
Pyright should also report a syntax error just like the python interpreter

**Comment by erictraut** [comment]
It looks like the python interpreter has a special case where it allows a generator without parens within an argument expression as long as there is only one argument. As soon as a second one is added to the argument list, this becomes a syntax error. The reasons for this limitation are not clear to me.

In any event, I agree that it would be good for pyright to emit an error here.

**Comment by erictraut** [comment]
Python 3.8 and newer also reject a single generator argument if there's a trailing comma after the argument expression.

I've added support to properly detect and report the runtime error. This will be included in the next release.

**Comment by erictraut** [comment]
This is included in pyright 1.1.215, which I just published. It will also be included in the next release of pylance.
